src/command_modules/azure-cli-eventhubs/azure/cli/command_modules/eventhubs/custom.py

```diff
@@ -104,28 +104,25 @@ def cli_eheventhub_update(instance, message_retention_in_days=None, partition_co
     if status:
         instance.status = status
 
-    if instance.enabled is True or enabled is True:
+    if enabled is True:
+        if instance.capture_description is None:
+            instance.capture_description = CaptureDescription()
+            instance.capture_description.destination = Destination()
+            instance.capture_description.encoding = EncodingCaptureDescription.avro
+
         if enabled:
             instance.capture_description.enabled = enabled
```

Comment:
You already check that `enabled` is `True` in the outer nesting. You can remove the `if` and move this up.

Reply:
resolved